src/profile/components/editProfile/EditProfile.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { loader } from 'graphql.macro';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+
+import styles from './EditProfile.module.css';
+import responsive from '../../../common/cssHelpers/responsive.module.css';
+import EditProfileForm, {
+  FormValues,
+} from '../editProfileForm/EditProfileForm';
+import { MyProfileQuery } from '../../graphql/__generated__/MyProfileQuery';
+import {
+  UpdateProfile as UpdateProfileData,
+  UpdateProfileVariables,
+} from '../../graphql/__generated__/UpdateProfile';
+import {
+  AddressType,
+  PhoneType,
+} from '../../../graphql/__generated__/globalTypes';
+import Explanation from '../../../common/explanation/Explanation';
+
+const UPDATE_PROFILE = loader('../../graphql/UpdateProfile.graphql');
+
+type Props = {
+  setEditing: () => void;
+  profileData: MyProfileQuery;
+};
+
+function EditProfile(props: Props) {
+  const { profileData } = props;
+  const { t } = useTranslation();
+  const [updateProfile, { loading }] = useMutation<
+    UpdateProfileData,
+    UpdateProfileVariables
+  >(UPDATE_PROFILE, {
+    refetchQueries: ['MyProfileQuery'],
+  });
+
+  const handleOnValues = (formValues: FormValues) => {
+    const variables: UpdateProfileVariables = {
+      profile: {
+        firstName: formValues.firstName,
+        lastName: formValues.lastName,
+        addPhones: [
+          !profileData?.myProfile?.primaryPhone?.id && formValues.phone
+            ? {
+                phone: formValues.phone,
+                primary: true,
+                phoneType: PhoneType.OTHER,
+              }
+            : null,
+        ],
+        updatePhones: [
+          profileData?.myProfile?.primaryPhone?.id
+            ? {
+                id: profileData?.myProfile?.primaryPhone?.id,
+                phone: formValues.phone,
+                primary: true,
+                phoneType: PhoneType.OTHER,
+              }
+            : null,
+        ],
+        addAddresses: [
+          !profileData?.myProfile?.primaryAddress?.id &&
+          (formValues.address || formValues.postalCode || formValues.city)
+            ? {
+                address: formValues.address,
+                city: formValues.city,
+                postalCode: formValues.postalCode,
+                primary: true,
+                addressType: AddressType.OTHER,
+              }
+            : null,
+        ],
+        updateAddresses: [
+          profileData?.myProfile?.primaryAddress?.id
+            ? {
+                id: profileData?.myProfile?.primaryAddress?.id,
```

Comment:
`id` is guaranteed to be something here because of the ternary check so it can be accessed directly without the optional chaining stuff `profileData.myProfile.primaryAddress.id`

Reply:
Fixed.